storages/backends/gcloud.py

```diff
@@ -158,34 +158,39 @@ def _save(self, name, content):
         name = self._normalize_name(cleaned_name)
 
         content.name = cleaned_name
-        file = GoogleCloudFile(name, 'rw', self)
+        file_object = GoogleCloudFile(name, 'rw', self)
 
         upload_params = {}
-        blob_params = self.get_object_parameters(name, content)
-        if 'cache_control' not in blob_params and self.cache_control:
-            warnings.warn(
-                'The GS_CACHE_CONTROL setting is deprecated. Use GS_OBJECT_PARAMETERS to set any '
-                'writable blob property or override GoogleCloudStorage.get_object_parameters to '
-                'vary the parameters per object.', DeprecationWarning
-            )
-            blob_params['cache_control'] = self.cache_control
+        blob_params = self.get_object_parameters(name)
+        upload_params['predefined_acl'] = blob_params.pop('acl')
 
-        upload_params['predefined_acl'] = blob_params.pop('acl', self.default_acl)
         if CONTENT_TYPE not in blob_params:
-            upload_params[CONTENT_TYPE] = file.mime_type
+            upload_params[CONTENT_TYPE] = file_object.mime_type
 
         for prop, val in blob_params.items():
-            setattr(file.blob, prop, val)
+            setattr(file_object.blob, prop, val)
 
-        file.blob.upload_from_file(content, rewind=True, size=content.size, **upload_params)
+        file_object.blob.upload_from_file(content, rewind=True, size=content.size, **upload_params)
         return cleaned_name
 
-    def get_object_parameters(self, name, content):
+    def get_object_parameters(self, name):
         """Override this to return a dictionary of overwritable blob-property to value.
 
         Returns GS_OBJECT_PARAMETRS by default. See the docs for all possible options.
         """
-        return self.object_parameters.copy()
+        object_parameters = self.object_parameters.copy()
+
+        if 'cache_control' not in object_parameters and self.cache_control:
+            warnings.warn(
+                'The GS_CACHE_CONTROL setting is deprecated. Use GS_OBJECT_PARAMETERS to set any '
+                'writable blob property or override GoogleCloudStorage.get_object_parameters to '
+                'vary the parameters per object.', DeprecationWarning
+            )
+            object_parameters['cache_control'] = self.cache_control
+        if 'acl' not in object_parameters:
+            object_parameters['acl'] = self.default_acl
+
+        return object_parameters
```

Comment:
The problem with putting this here is that people will override this method. Then the ACL won’t be set unless they call `super`.

Reply:
I added the `default_acl` logic to the code areas where object_parameters is called from.